labs/b5.md

```diff
@@ -0,0 +1,364 @@
+---
+title: Lab 5 - Networking 101
+layout: lab
+---
+
+## Overview
+It is undeniable that the internet is an important system that has redefined our world. The ability to develop networks and allow devices to communicate is critical to value of modern day computer systems.  This lab will take a look into the basics of computer networking and then examine networks through the perspective of a sysadmin.
+
+We will be using web browsing as an analogy to understand the basics of networking. What exactly happens when I go web browsing for cat pictures?
+
+But first let's take a short dive into the details of networking.
+
+---
+
+## OSI Model
+To lay the groundwork for understanding networking, we'll first turn to the Open Systems Interconnection model (OSI).  The OSI model describes modern computer systems by partitioning them into several 'layers':
+1. **Physical Layer**
+This layer deals with the physical transmission of the data such as passing electrical signals over a fiber optic cable or radio frequencies for wireless.
+2. **Data Link Layer**
+This layer deals with transfer data between network nodes in a wide area network (WAN) or a local area network (LAN). An example of this is L2 routing according to MAC addresses.
+3. **Network Layer**
+This layer deals with packet forwarding and routing through intermediate routers. The most common L3 protocol, and the one you're probably familiar with, is Internet Protocol (IP).  This layer is concerned with delivering data between hosts that correspond to IP addresses but makes no guarantees about reliable transport of the packets.
+4. **Transport Layer**
+This layer is responsible for providing reliable transport, and reassembling packets that may have arrived out of order.  Protocols in this layer provide host-to-host communication services for applications. The most well known protocols are TCP (connection-oriented) and UDP(connection-less).
```

Comment:
should probably briefly explain what "reliable transport" means and connection-oriented vs connection-less.

Reply:
I left the details for the more detailed section further into the lab -- will remove the mentions here to defer explanations to that portion.